apis/Google.Cloud.Logging.Console/Google.Cloud.Logging.Console.Tests/GoogleCloudConsoleFormatterTest.cs

```diff
@@ -152,6 +153,101 @@ public void ConsoleLoggerOptions_OptionChange_IsReloaded()
         Assert.Equal(expectedJsonWithScope, actualJsonWithScope);
     }
 
+    [Fact]
+    public void ConsoleLoggerOptions_EmptyProjectId_TraceInformationIsNotLogged()
+    {
+        // Don't set the TraceGoogleCloudProjectId.
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions());
+
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+        var writer = new StringWriter { NewLine = "\n" };
+
+        formatter.Write(logEntry, scopeProvider: null, writer);
+
+        var expectedJson = "{\"message\":\"test\",\"category\":\"LogCategory\",\"severity\":\"INFO\"}\n";
+        var actualJson = writer.ToString();
+
+        Assert.Equal(expectedJson, actualJson);
+    }
+
+    [Fact]
+    public void Log_NullActivity_TraceInformationIsNotLogged()
+    {
+        // Set the TraceGoogleCloudProjectId.
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions { TraceGoogleCloudProjectId = "my-project-id" });
+
+        // Set Activity.Current to null for clarity. (It is already null here as no Activity is created.)
+        Activity.Current = null;
+
+        var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+        var writer = new StringWriter { NewLine = "\n" };
+
+        formatter.Write(logEntry, scopeProvider: null, writer);
+
+        var expectedJson = "{\"message\":\"test\",\"category\":\"LogCategory\",\"severity\":\"INFO\"}\n";
+        var actualJson = writer.ToString();
+
+        Assert.Equal(expectedJson, actualJson);
+    }
+
+    [Fact]
+    public void Log_DefaultTraceId_TraceInformationIsNotLogged()
+    {
+        // Set the TraceGoogleCloudProjectId.
+        var formatter = CreateFormatter(new GoogleCloudConsoleFormatterOptions { TraceGoogleCloudProjectId = "my-project-id" });
+
+        // Backup the generator, so that it can be reset after the test.
+        // It is null by default but just to keep it easy to understand.
+        var generator = Activity.TraceIdGenerator;
+
+        // Simulate the condition where the generated `TraceId` is equal to the default value of `ActivityTraceId`.
+        Activity.TraceIdGenerator = () => default;
+        using var activity = new Activity("my-activity").Start();
+
+        // Reset the generator, so that other tests aren't impacted.
```

Comment:
I'd put this in a finally block, to avoid issues if either of the previous lines throws an exception.

Reply:
And if tests are running in parallel, this will be flaky.

But, maybe after my comment on IsStopped, we don't need this test anymore.